### ISSUE TITLE
PostSonarr

### ISSUE THREAD

**Issue by TinTin192** [mdhiggins/sickbeard_mp4_automator#1021]
Hi there.
I've just updated Sonarr to V3 (phantom-develop) and upon getting everything back into place. 
Obviously i want my script's to work.

Nothing has changed within any files/system etc but i get an error when setting up the script.

  | SonarrErrorPipeline | Invalid request Validation failed:   -- Script exited with code: 1 
 
  | postSonarr.py | TypeError: int() argument must be a string or a number, not 'NoneType' 

  | postSonarr.py | season = int(os.environ.get('sonarr_episodefile_seasonnumber')) 

  | postSonarr.py | File "/home/(removed)/.scripts/postSonarr.py", line 34, in <module>  

  | postSonarr.py | Traceback (most recent call last): 

  | SonarrErrorPipeline | Invalid request Validation failed:   -- Script exited with code: 1

Anyone got an idea why this is happening

**Comment by mdhiggins** [comment]
Believe there was an update to address this, you'll just need to update

**Comment by n0cebo** [comment]
Updated...still have the same issue

**Comment by strafer** [comment]
I've got the same "exit code 1" error with Sonarr version 3.

**Comment by mdhiggins** [comment]
Does your postSonarr.py script have the sys.exit(0) line at the end?
Additionally is there anything in the log files to give more info about the error? Line 34 doesn't have anything to do with exit codes so this whole thing is strange.

**Comment by strafer** [comment]
> Does your postSonarr.py script have the sys.exit(0) line at the end?
> Additionally is there anything in the log files to give more info about the error? Line 34 doesn't have anything to do with exit codes so this whole thing is strange.

Yes the sys.exit(0) line is at the end. The Sonarr log has the following error:

`19-7-1 11:17:57.9|Debug|CustomScript|Executed external script: /Users/user/Scripts/sickbeard_mp4_automator/postSonarr.py - Status: 1
19-7-1 11:17:57.9|Debug|CustomScript|Script Output: 
7/1/2019 6:17:57 PM - Error - Traceback (most recent call last):
7/1/2019 6:17:57 PM - Error -   File "/Users/user/Scripts/sickbeard_mp4_automator/postSonarr.py", line 5, in <module>
7/1/2019 6:17:57 PM - Error -     import requests`

The requests module is installed in in /usr/local/lib/python2.7/site-packages (2.22.0) but when I check "which python" it returns /usr/local/bin/python and pip is also in /usr/local/bin. I'm not sure why pip is installing in /usr/local/lib/python2.7 instead of /usr/local/bin/python. I'm on macOS Mojave.

**Comment by strafer** [comment]
I nuked my entire Homebrew install, reinstalled Homebrew, and then installed Python2, Pip, and SetupTools, but Sonarr is still giving me "ImportError: No module named requests". I don't know what to do at this point.

**Comment by mdhiggins** [comment]
It seems like sonar is probably using a different python environment for some reason

Do you not have your custom script configured correctly and pointing to the right python environment?

Post a pic of your sonar custom script settings

**Comment by strafer** [comment]
Doesn't the custom script in Sonarr point to the postSonarr.py script?

[Screenshot](https://www.dropbox.com/s/8knebbjhxmlhyy4/Screenshot%202019-07-01%2013.16.30.png?dl=0)

**Comment by mdhiggins** [comment]
This is how mine is setup

![image](https://user-images.githubusercontent.com/3608298/60464233-f6ca1680-9c1b-11e9-8dd2-bfcb9dd870a5.png)

You should probably aim it to the correct python executable/environment and then add the path to the script. If they got rid of the argument box on Sonarr 3 which is looks like they might have I would just add it on one line like so

`/usr/local/bin/sma/env/bin/python3 /pathtoscript/postSonarr.py`

That might work

**Comment by strafer** [comment]
Unfortunately, your suggestion doesn't work. It says "file does not exist." What's weird, my previous setup with Sonarr 2 and sickbeard_mp4_automator worked fine with the same setup in my screenshot. I never had to specify the python path.

I might try downgrading to Sonarr 2 on this computer to see if it works.

**Comment by mdhiggins** [comment]
Just wrap that one line into a shell script and point it to that instead if that doesn't work and pass the command line parameters

`/usr/local/bin/sma/env/bin/python3 /pathtoscript/postSonarr.py "$@"` I think

**Comment by strafer** [comment]
That might be a bit beyond my skill level. I've created a bash script and pasted in that line, but I don't know how to "wrap that one line." Is there a character I need to insert between the two paths?

**Comment by mdhiggins** [comment]
Just paste that line as is

**Comment by strafer** [comment]
Okay, tested it with the script and got the same error as before:

`ImportError: No module named requests
19-7-1 14:14:57.2|Warn|SonarrErrorPipeline|Invalid request Validation failed:`

**Comment by strafer** [comment]
Homebrew's python and PIP wouldn't be the source of the problem, would it?

**Comment by mdhiggins** [comment]
Tough to say, you just need to make sure whichever python environment you’re installing your dependencies on is the same one you’re executing from your script. It might be the problem

**Comment by strafer** [comment]
I just tested it in Sonarr version 2 on the same computer and it worked perfectly. So it looks like it's a Sonarr3 issue with the beta. Maybe it's incomplete or broken.

**Comment by mdhiggins** [comment]
I'm thinking Sonarr V3 must either natively execute the program within its own python environment.  That being said I still think the solution for V3 is to just ensure you're executing from the right one. The docker containers just use their own virtual environment to do this which is the cleanest solution.

**Comment by strafer** [comment]
I solved the postsonarr.py script failure in Sonarr v3. I had to change the python path in the script to "/usr/local/bin/python2.7" or "/usr/local/bin/python2" for the script to work. Even though "which python" and "/usr/bin/env python" both point to the correct python (Python 2.7.16), the script will fail if the path at the start is "/usr/local/bin/python" or "/usr/bin/env python".

I wonder if I should hard-code all the python paths by doing a search and replace of the entire mp4_automator folder?
